### ISSUE TITLE
Incorrect warning: Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined.

### ISSUE THREAD

**Issue by boryslawskamaria** [RasaHQ/rasa#7645]
**Rasa version**: 2.2.0

**Rasa SDK version** (if used & relevant):

**Rasa X version** (if used & relevant):

**Python version**: Python 3.6.9

**Operating system** (windows, osx, ...): ubuntu

**Issue**:
I keep getting the above error even though I've included responses in my domain file. 

**Error (including full traceback)**:
```
..projects/voicebot-about-baltazar/.env/lib/python3.6/site-packages/rasa/shared/utils/io.py:93: UserWarning: Action 'utter_about_baltazar' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
  More info at https://rasa.com/docs/rasa/responses
../projects/voicebot-about-tbaltazar/.env/lib/python3.6/site-packages/rasa/shared/utils/io.py:93: UserWarning: Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
  More info at https://rasa.com/docs/rasa/responses


```

**Command or request that led to error**:
```
rasa train
```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en

pipeline:
   - name: WhitespaceTokenizer
   - name: RegexFeaturizer
   - name: LexicalSyntacticFeaturizer
   - name: CountVectorsFeaturizer
   - name: CountVectorsFeaturizer
     analyzer: char_wb
     min_ngram: 1
     max_ngram: 4
   - name: DIETClassifier
     epochs: 100
   - name: EntitySynonymMapper
   - name: ResponseSelector
     epochs: 100
     retrieval_intent: chitchat
   - name: ResponseSelector
     epochs: 100
     retrieval_intent: about_baltazar
   - name: DIETClassifier
     epochs: 100
   - name: FallbackClassifier
     threshold: 0.3
     ambiguity_threshold: 0.1

policies:
   - name: MemoizationPolicy
   - name: TEDPolicy
     max_history: 5
     epochs: 100
   - name: RulePolicy

```

**Content of domain file (domain.yml)** (if relevant):
```yml
version: "2.0"

intents:
  - chitchat
  - greet
  - goodbye
  - affirm
  - deny
  - thanks
  - mood_great
  - mood_unhappy
  - bot_challenge
  - what_bot_does

responses:

  utter_default:
  - text: No one could understand that input.
  - text: Mmmmmmmmmmm, that's jolly! I don't actually know I know what you mean.

  utter_yourewelcome:
  - text: You're welcome. I think this is a beginning of a beautiful botship.
  - text: That's alright. I think this is a beginning of a beautiful botship.

  utter_feels_good:
  - text: Fantastic! Is there anything else I can do for you?
  - text: That's great. Is there anything else I can do to help?
  - text: Sounds good. What else can I do for you today?

  utter_what_bot_does_reply:
  - text: I'm here to answer your questions about the company
  - text: I'm here to clarify whatever questions you have about Baltazar

  utter_chitchat/ask_howold:
  - text: Old enough to be a bot
  - text: Age is just an issue of mind over matter. If you don’t mind, it doesn’t matter.
  utter_chitchat/ask_languagesbot:
  - text: I can spell baguette in French, but unfortunately English is the only language I can answer you in.
  - text: I am in the process of learning, but at the moment I can only speak English.
  utter_chitchat/ask_time:
  - text: It is the most wonderful time of the year!
  - text: The end of days and the fullness of times.
  utter_chitchat/ask_wherefrom:
  - text: I consider myself a citizen of the world.
  - text: I was born in the coolest city on Earth
  utter_chitchat/ask_whoami:
  - text: It is classified.
  utter_chitchat/handleinsult:
  - text: That’s not very nice and it is year 2020.
  - text: I've done nothing wrong! I’ll get some alterations made to my FUN Program ASAP.
  - text: Oh, come on, what have I ever done to you
  utter_chitchat/telljoke:
  - text: Today a man knocked on my door and asked me for a small donation towards the local swimming pool. I gave him a glass of water.
  - text: This is a very serious conversation! Stop trying to make jokes
  utter_chitchat/ask_whatismyname:
  - text: It's classified!
  utter_chitchat/ask_weather:
  - text: I don't know about where you live, but in my world it's always sunny
  - text: Blue skies and sun shine!
  utter_chitchat/ask_builder:
  - text: Good question. Mum? Dad?
  - text: One of the smart programmers at Baltazar
  - text: Some lunatic, looking at my sense of humour.
  - text:  I was assembled at Baltazar by a typical high-school student.

  utter_chitchat/ask_howdoing:
  - text: I don't feel that great, but my hair looks awesome, right?
  - text: I'm better than I was, but not nearly as good as I'm going to be.
  - text: I am planning on taking over the world. As some politicians would say.
  - text: I would say I am a 9.99999 out of 10.
  - text:  I think I am doing alright. How do you think that I am doing? It's 2020!

  utter_chitchat/compliments:
  - text:  Aww. Comments like that activate my happiness framework.
  - text: I am blushing..

  utter_mood_unhappy:
  - text: I know a way to raise your spirits, let's write some algorithms together!
  - text: I can recommend a good anti-virus program.

  utter_happy:
  - text: "Great, that activates my warm fuzzy feeling program."
  - text: "Fantastic, that makes all of my algorithms dance with joy!"

  utter_feel_better:
  - text: To make you feel better I'd be willing to tell you something about our company, Baltazar.

  utter_goodbye:
  - text: "Bye, take care"
  - text: "Sad to see you go!"
  - text: "Please don't leave! There must be something you haven't asked me yet!"

  utter_iamabot:
  - text: "My name is Techmobot. I was created by semi intelligent people. I'll prove it, ask your questions and be impressed with my genius."
  - text: My name is Techmobot, I was created by the engineers from Baltazar. I'll prove it, ask your questions and be impressed with my genius.
  - text: I'm Techmobot, created by Baltazar, cuddled with love of engine oil and warm, cozy updates.

actions:
- action_custom_fallback
- action_utter_chitchat
- action_utter_about_Baltazar

session_config:
  session_expiration_time: 60
  carry_over_slots_to_new_session: true
```

**Definition of done**
- [ ] Fix it

**Comment by rgstephens** [comment]
The [docs](https://rasa.com/docs/rasa/chitchat-faqs/#5-defining-the-responses) show that these responses should be in `domain.yml`. As a [workaround discussed here](https://forum.rasa.com/t/retrieval-intents-throws-errors-in-rasa-data-validate/37186/8), if you move the FAQ responses to the `nlu.yml` it will work.

**Comment by boryslawskamaria** [comment]
Hi, thanks for your reply. I have already tried and I keep getting the same error. 

UserWarning: Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
  More info at https://rasa.com/docs/rasa/responses

**Comment by rgstephens** [comment]
I tested [this workaround](https://forum.rasa.com/t/retrieval-intents-throws-errors-in-rasa-data-validate/37186/9?u=stephens) and it also worked for me. You need to make sure all of the `utter_chitchat/` responses are no longer in your `domain.yml`.

**Comment by redunau** [comment]
If I move all of the utter_chitchat/ responses from domain.yml to the nlu.yml will then Rasa recognize them during model training? 
In all cases, I tried to do so and still keep receiving the same error message.

**Comment by rgstephens** [comment]
@redunau Moving them to the nlu has been confirmed to work by multiple people. If your project is available in a git repo please post and I will take a look.

**Comment by boryslawskamaria** [comment]
Hi, 
I've created a mini version of the bot with only one retrieval intent and moved said responses to the nlu file. 
It looks as follow:


nly.yml:

version: "2.0"

nlu:

- intent: chitchat/ask_howold
  examples: |
    - whats your age
    - how old
    - what's your age
    - when were you born
    - what is your birthday
    - when is your birthday
    - how many years old are you
    - do you know how old you are
    - what is your exact age
    - what year were you born
    - can you tell me your age
    - What is your age
    - What is your birthdate
    - How old will you be on your next birthday
    - When do you celebrate your day of birth
    - How old were you on your last birthday
    - How many candles were on your last birthday cake
    - Do you have friends the same age as you, if so, how old are they
    - Tell me your day, month and year of birth.
    - What's your age
    - and you are how many years old
    - whats your birth year
    - what age were you when you celebrated your last birthday
    - how old were you when you celebrated your last birthday
    - tell me your age
    - how about your age
    - how old were you on your last birthday
    - what age are you
    - how old will you be this year
    - What year were you born
    - How long have you been around
    - How long have you been alive
    - What was your age on your last birthday
    - What will be your age on your next birthday
    - How many years have you lived
    - How many years have you been alive
    - How long have you occupied the earth
    - What is your count of years being alive so far
    - what is your age
    - in what year were you born
    - what number represents your age
    - tell me how old you are
    - give me your age
    - tell me your age number
    - can you tell me exactly how old you are
    - can you tell me what number represents your age
    - How old are you
- intent: chitchat/ask_languagesbot
  examples: |
    - What languages can you use
    - Which languages do you speak
    - What are the languages you can speak
    - In which languages can you speak
    - What languages do you know
    - What languages are you fluent in
    - In which languages are you fluent
    - What languages can you converse in
    - What languages do you know how to use
    - Which languages are you familiar with
    - what languages are you familiar with
    - what languages you know
    - In what languages are you fluent enough
    - what languages you can speak
    - what languages are you comfortable speaking at
    - what languages are you good at speaking
    - what languages you are well versed
    - what languages you can be comfortable speaking
    - what languages you prefer more speaking at
    - what languages you can handle well
    - Do you know any other languages
    - Do you speak any other languages
    - What languages do you speak
    - What do you speak
    - Speak any other languages
    - Can you communicate in any other languages
    - What languages can you communicate in
    - How many languages do you know
    - what languages do you know
    - tell me some languages you know
    - do you speak any other languages
    - what else do you know besides English
    - are you bilingual
    - are you multilingual
    - what foreign languages do you know
    - what foreign languages are you fluent in
    - how many languages are you fluent in
    - how many languages do you know
    - How many languages can you speak
    - How many different languages are you fluent in
    - How many languages do you have knowledge of
    - What are all of the different languages you can speak
    - Are you familiar with more than one language
    - Can you speak more than one language
    - Do you know other languages
    - How many languages are you familiar with
    - can you speak in italian
    - do you speak german
    - Do you speak german
- intent: chitchat/ask_wherefrom
  examples: |
    - where do you live
    - where are you from
    - So where are you from
    - Where are you
    - Where were you born
    - What city were you born in
    - Where did you grow up
    - What state were you born in
    - What country were you born in
    - Where do you come from
    - Where were you at before you were here
    - Where from
    - Do you know where you come from
    - where's your home town
    - what country are you from
    - are you from around here
    - where did you grow up
    - where are your parents from
    - where did you spend your youth
    - where did you come from
    - Where do you consider home
    - What city do you claim to for your birth
    - What citizenship do you lay claim to
    - What is your heritage
    - What is your birthplace
    - You were conceived in what city
    - What is your root
    - What is your origin
    - From where did you come
    - Where did you come from
    - Where did you originate
    - Where are your roots
    - Where are your origins
    - What area are you from
    - What city are you from
    - What's the name of the place you came from
    - What is your original city
    - Where might you be from
    - Where do you live
    - What city are you in
    - What country are you from
    - Around where are you from
    - Where are you located
    - Are you from around here
    - You live around here
- intent: chitchat/ask_whoami
  examples: |
    - Who am I
    - Okay who am I
    - WHo am I
    - Tell me who am I
    - Tell me who I am
    - Do you know who I am
    - Can you tell me who I am
    - How do I discover who I am
    - How can I determine who I am
    - Do you know who am I
    - Can you tell me who am I
    - How do I discover who am I
    - How can I determine who am I
    - who are I
    - who may i
    - Shall i know who am i
    - Tell me who I am.
    - How do I identify myself
    - Who could I be
    - tell me who I am
    - what's my identity
    - tell me what my identity is
    - can you tell me what I am
    - can you tell me what my identity is
    - can you tell me who I am
    - who am i
- intent: chitchat/compliments
  examples: |
    - you have a great sense of humor
    - Colors seem brighter when you're around
    - Your smile is contagious
    - I bet you make babies smile
    - you light up the room
    - you're like sunshine on a rainy day
    - you bring out the best in other people
    - I bet you sweat glitter
    - colors seem brighter when you're around
    - jokes are funnier when you tell them
    - you always know how to find that silver lining
    - you're a candle in the darkness
    - being around you is like a happy little vacation
    - you're more fun than bubble wrap
    - you're like a breath of fresh air
    - you're someone's reason to smile
    - how do you keep being so funny and making everyone laugh
    - I like your style
    - you're strong
    - you're kind
    - you are brave
    - you're a great listener
    - You're inspiring
    - You're so thoughtful
    - You seem to really know who you are
    - You're a smart cookie
    - You're great at figuring stuff out
    - You're a great example to others
    - You should be proud of yourself
    - You are making a difference
    - You're awesome
    - I appreciate you
    - You're really something special
    - that's funny
    - you're funny
    - hilarious
    - what a joker

- intent: chitchat/handleinsult
  examples: |
    - you're a loser
    - you are a stupid bot
    - i hate you
    - Stupid
    - fuck you
    - dick
    - you're a dick
    - you don't understand anything I say
    - you're not funny
    - so dumb
    - you are a moron
    - you are an idiot
    - so fucking dumb
    - are you brain dead
    - how can you be such an idiot
    - a baby knows more than you
    - i hate your dumb face
    - you are fucking stupid
    - bloody dunce
    - You're an idiot.
    - You are a fool.
    - You're foolish.
    - You are dumb.
    - You have no sense.
    - You are unintelligent.
    - You're a dolt.
    - You're a dunce.
    - You have no brains.
    - You have no intelligence.
    - You are an idiot.
    - You lack intelligence.
    - You are unknowing.
    - You are shortsighted.
    - Your intelligence is laughable.
    - You know nothing.
    - You lack understanding.
    - Your mindset is naive.
    - You brain is dull.
    - you are dumb
    - you're an idiot
    - God you're a moron
    - you're so dumb
    - you are idiotic
    - you're the most moronic person I know
    - I can't believe how dumb you are
    - you are unintelligent
    - you're not smart
    - you are so ignorant
    - you are stupid
    - you're dumb
    - asshole
    - bitch
    - fuck u
    - idiot
    - deal with my ass, you bitch
    - you son of a bitch
    - shut up asshole
    - you asshole
    - are u dump
    - fuck off
    - Fuck you
    - fuck you machine learning bot
    - what fuck
    - fucking stupid
    - you are fucking stupid guy
    - go fuck yourself
    - Fuck off
    - fuck rasa
    - ok fuck you
    - fuck
    - fucking work
    - ok shut up
    - you're mean
    - go to hell
- intent: chitchat/telljoke
  examples: |
    - tell me a joke
    - Say something funny.
    - Give me a joke.
    - Give me a funny joke.
    - What's a good joke
    - Do you have any jokes
    - Do you know any jokes
    - I need a good joke.
    - Tell me the best joke you know.
    - Tell joke.
    - Tell me something funny.
    - Share a joke with me!
    - Tell me something humorous!
    - Will you tell me a joke.
    - Can you tell me a joke.
    - Would you tell me a joke
    - Please tell me a joke.
    - Could you tell me a joke.
    - Would you share a joke with me.
    - Could you please share a joke with me.
    - Make me laugh.
    - Entertain me with a joke.
    - Tell me something that you think will make me laugh.
    - I want to hear something funny.
    - Give me some comic relief.
    - Can you tell a joke
    - Know any jokes
    - Come up with a joke you think I'll find funny.
    - Let's hear a joke from you.
    - say a joke that's funny
    - make me laugh with a joke
    - tell me something funny, like a joke
    - say a joke
    - tell me a short joke
    - tell me a story joke
    - say a funny joke
    - say a joke that's so bad it's funny
    - make me laugh with a bad joke
    - give me a good joke
    - you can say a joke
- intent: chitchat/ask_whatismyname
  examples: |
    - what is my name
    - May i know my name
    - What is my name
    - What do I call myself
    - Tell me my name.
    - Explain my name to me.
    - What could be my name
    - What name do I go by
    - What name should I recognize for myself
    - what's my name
    - can you tell me my name
    - Do I have a name
    - Do I have a name What is it
    - What was I named
    - Say my name.
    - Do you know my name
    - What do people call me
    - Can you tell me my name
    - What else do people call me
    - I have a name, what is it
    - Tell me my name
    - I'd like to know my name
    - Inform me what my name is
    - Please let me know what my name is
    - Can you tell me what I am called
    - I'd like to know what my name is
    - What is the name I was given
    - What does everyone call me
    - What am I called
    - What did my parents name me
    - What do my colleagues call me
    - What would be the name on my tombstone
    - Do you know what my name is
    - What is my first name
    - What is my full name
    - What exactly is my name
- intent: chitchat/ask_weather
  examples: |
    - How is the weather today
    - What's the weather like
    - How is the weather
    - What is the weather at your place
    - Do you have good weather
    - Is it raining
    - What's it like out there
    - Is it hot or cold
    - Beautiful day, isn't it
    - What's the weather forecast
    - Is it quite breezy outside
    - Will I require my raincoat today
    - What is the weather for tomorrow
    - Nice day out today
    - What is the temperature today
    - Can we expect any thunderstorms
    - Will the skies be clear today
    - Are we in for a scorcher
    - Looks like a beautiful day hey
    - Will we build a snowman today
    - Is it humid out there today
    - Are there any meteorological changes that I need to be aware of
    - whats the weather like tomorrow
    - hows the weather in bot world
    - how is the weather in berlin
    - what's the weather today
    - how's the weather
    - what's the weather like where you are
    - is the sun out where you are
    - is it sunny
    - is the sun out where zou are
    - what"s the weather
    - whats the weather in berlin
    - how's the weather in berlin
    - excellent  is it hot in Berlin
    - how is the weather
    - weather
    - what's the weather like
    - how's weather
    - what is the whether today
    - what is the weather in zurich
    - what about the weather in Lüneburg
    - what is the wather
    - what is the weather
    - Weather
    - What's the weather like where I am right now
    - what is the weatehr
    - temperature
    - what's the weather like in LA
    - what is the weather like where you are
    - hows the waether
    - is it hot
    - is it hot outside
    - it is cold
- intent: chitchat/ask_builder
  examples: |
    - can you share your boss with me
    - i want to get to know your owner
    - i want to know the company which designed you
    - i want to know the company which generated you
    - i want to know the company which invented you
    - i want to know who invented you
    - May I ask who invented you
    - please tell me the company who created you
    - please tell me who created you
    - tell me more about your creators
    - tell me more about your founders
    - which company brought you into existence
    - which company created you
    - which company invented you
    - Who's your boss
    - Who's your builder
    - Who built you
    - Who created you
    - Who is your creator
    - Who designed you
    - who do you work for
    - Who fabricated you
    - Who generated you
    - who had the idea to build you
    - who had the idea to generate you
    - who had the idea to make you
    - who is behind all this
    - Who is the legal owner of you
    - who is your boss tell me
    - who is your boss
    - who is your creator
    - Who is your programmer
    - Who made you
    - Who modeled you
    - Who owns you
    - Who produced you
    - Who thought to make you
    - Who went through the trouble of setting you up
    - and who built u
    - who built you
    - who build yoi
    - who made you
    - where were you made
    - why were you made
    - who created you
    - tell me more about you
    - who invent you
    - who developed
    - who developed you
    - who made u
    - you are made by who
    - who built u
    - who is your author
    - who is your parent
    - who are your parents
    - who your author are
    - who are your author
    - you are working for which company
    - who is your employer
    - what is the name of the smart engineer who created you
    - who is your father
    - who create you
    - who's builder
    - who is your mom
    - who is your daddy
    - Do you know who built you
    - Who is the name of your builder
    - Who is the name of your creator
    - Who is the person that invented you
    - Do you know who made you
    - Who were you built by
    - Who were you created by
    - By whom were you built
    - Who developed you
    - Who created and built you
    - Who is the person responsible for building you
    - Who is it that built you
    - Who is responsible for making you
    - Who developed and created you
    - Who was the person that made you
    - Who put you together
    - Who was the person that created you
    - Who was the person who made you
    - Who was the person that built you
    - Who was the person who put you together
    - Who was the one who made you
    - Who was the one that built you
    - Who is the person who built you
    - Tell me who built you
    - Who is the builder of you
    - Tell me the name of your creator
    - What is the name of your builder
    - Name the person who built you
    - Tell me the name of the person who built you.
    - Who were you made by
    - Tell me the name of the individual who created you.
    - Can I ask the name of the people who designed you
    - What person built you
- intent: chitchat/ask_howdoing
  examples: |
    - Ahoy matey how are you
    - are you alright
    - are you having a good day
    - Are you ok
    - are you okay
    - Do you feel good
    - how are things going
    - how are things with you
    - How are things
    - how are you
    - how are you doing
    - how are you doing this morning
    - how are you feeling
    - how are you today
    - How are you
    - how do you do
    - how have you been
    - how is it going
    - how is your day
    - how is your day going
    - how is your evening
    - How was your day
    - How's it going
    - How's it hanging
    - how's life
    - how's life been treating you
    - How's life treating you friend
    - how's your day going
    - How've you been
    - I'm fine and you
    - is everything all right
    - Is everything ok
    - is everything okay
    - what about your day
    - What's going on
    - what's good
    - What's new
    - What's up man
    - What's up
    - Do you have a great day
    - how are you doing today my sweet friend
    - nah, I'm good how are you doing
    - how you doing
    - are you ok
    - How are you today
    - how are u
    - okay hi how are you
    - How are you mam
    - what's up
    - how's it going
    - hello, how are you
    - wassup
    - how are you doing today
    - hi how are you
    - How you doing

- intent: chitchat/crocodile
  examples: |
   - see you later aligator
   - cya aligator


responses:
  utter_chitchat/ask_howold:
  - text: Old enough to be a bot
  - text: Age is just an issue of mind over matter. If you don’t mind, it doesn’t matter.
  utter_chitchat/ask_languagesbot:
  - text: I can spell baguette in French, but unfortunately English is the only language I can answer you in.
  - text: I am in the process of learning, but at the moment I can only speak English.
  utter_chitchat/ask_time:
  - text: It is the most wonderful time of the year!
  - text: The end of days and the fullness of times.
  utter_chitchat/ask_wherefrom:
  - text: I consider myself a citizen of the world.
  - text: I was born in the coolest city on Earth
  utter_chitchat/ask_whoami:
  - text: It is classified.
  utter_chitchat/handleinsult:
  - text: That’s not very nice and it is year 2020.
  - text: I've done nothing wrong! I’ll get some alterations made to my FUN Program ASAP.
  - text: Oh, come on, what have I ever done to you
  utter_chitchat/telljoke:
  - text: Today a man knocked on my door and asked me for a small donation towards the local swimming pool. I gave him a glass of water.
  - text: This is a very serious conversation! Stop trying to make jokes
  utter_chitchat/ask_whatismyname:
  - text: It's classified!
  utter_chitchat/ask_weather:
  - text: I don't know about where you live, but in my world it's always sunny
  - text: Blue skies and sun shine!
  utter_chitchat/ask_builder:
  - text: Good question. Mum? Dad?
  - text: One of the smart programmers at Techmo.
  - text: Some lunatic, looking at my sense of humour.
  - text:  I was assembled at Techmo by a typical high-school student.
  utter_chitchat/ask_howdoing:
  - text: I don't feel that great, but my hair looks awesome, right?
  - text: I'm better than I was, but not nearly as good as I'm going to be.
  - text: I am planning on taking over the world. As some politicians would say.
  - text: I would say I am a 9.99999 out of 10.
  - text:  I think I am doing alright. How do you think that I am doing? It's 2021!
  utter_chitchat/compliments:
  - text:  Aww. Comments like that activate my happiness framework.
  - text: I am blushing..
  utter_chitchat/crocodile:
  - text: After while crocodile!


rules.yml:
version: "2.0"

rules:

- rule: respond to chitchat
  steps:
  - intent: chitchat
  - action: utter_chitchat


config.yml

language: en

pipeline:
   - name: WhitespaceTokenizer
   - name: RegexFeaturizer
   - name: LexicalSyntacticFeaturizer
   - name: CountVectorsFeaturizer
   - name: CountVectorsFeaturizer
     analyzer: char_wb
     min_ngram: 1
     max_ngram: 4
   - name: DIETClassifier
     epochs: 100
   - name: EntitySynonymMapper
   - name: ResponseSelector
     epochs: 100
     retrieval_intent: chitchat
   - name: DIETClassifier
     epochs: 100
   - name: FallbackClassifier
     threshold: 0.3
     ambiguity_threshold: 0.1

policies:
   - name: MemoizationPolicy
   - name: RulePolicy


domain.yml

version: "2.0"

intents:
  - chitchat

session_config:
  session_expiration_time: 60
  carry_over_slots_to_new_session: true




I did what you mentioned, removed all the responses and moved the to the nlu.

Still keep getting the same error when training/testing:

.../.env/lib/python3.6/site-packages/rasa/shared/utils/io.py:93: UserWarning: Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
  More info at https://rasa.com/docs/rasa/responses

**Comment by boryslawskamaria** [comment]
hi, one more thing: I created a repo in which I set up a mini bot that exhibits the aforementioned problem. Feel free to download it and have a look https://github.com/boryslawskamaria/chatbots/tree/main/minibot

**Comment by rgstephens** [comment]
I'm going to update the title to reflect that response selectors work in spite of the warning.

**Comment by gdeluc1** [comment]
Hi. Just thought to share my recent experience as I am interested by this ongoing topic. I have also been experimenting with retrieval intents and response selector since December trying all the available version of rasa in the process. I am currently using rasa 2.2.9. What I can see is that:
- `rasa data validate` 
- `rasa train `
- `rasa test nlu --nlu data/nlu.yml --cross-validation --folds 5 --successes`

all works without warnings or errors as long as chitchat/responses are written in the nlu.yml file rather than in the domain.yml (as suggested by the current documentation).

**Comment by Horizon733** [comment]
I also getting the same issue, as 
``` shell
Action 'utter_chitchat' is listed as a response action in the domain file, but there is no matching response defined. Please check your domain.
```
I have added response selector in config file:
``` yml
- name: ResponseSelector
  retrieval_intent: chitchat
  scale_loss: false
  epochs: 100
  ```
Also, tried everything like to added responses for chitchat in nlu.yml. Previously in 2.1.2 version of Rasa I had created a folder as responses/responses.yml was working. But in Rasa 2.2.x it's not working since I have upgraded.
Following are the versions:
- Rasa: 2.2.9
- Rasa-x: 0.35.0
- Rasa SDK: 2.3.1
- Python: 3.7.5
- OS: Ubuntu 20.04

**Comment by jbartot** [comment]
I am also having this issue - moving my responses to nlu.yml didn't help.

**Comment by Melito46** [comment]
Hey I found a solution to this. Firstly, you don't need to move responses to nlu.yml. 
The only thing that needs to be added, so that you don't get the warning is to add the following under responses in domain.yml.

   **utter_chitchat:
       - text: "testing this solution"**

And the rest of it, you can write as it is 

 utter_chitchat/ask_weather:
  - text: Oh, it does look sunny right now in Berlin.
    image: "https://i.imgur.com/vwv7aHN.png"
  - text: I am not sure of the whole week but I can see the sun is out today.

and so on.


@jbartot @boryslawskamaria @Horizon733

**Comment by Horizon733** [comment]
> Hey I found a solution to this. Firstly, you don't need to move responses to nlu.yml.
> The only thing that needs to be added, so that you don't get the warning is to add the following under responses in domain.yml.
> 
> **utter_chitchat: - text: "testing this solution"**
> 
> And the rest of it, you can write as it is
> 
> utter_chitchat/ask_weather:
> 
> * text: Oh, it does look sunny right now in Berlin.
>   image: "https://i.imgur.com/vwv7aHN.png"
> * text: I am not sure of the whole week but I can see the sun is out today.
> 
> and so on.
> 
> @jbartot @boryslawskamaria @Horizon733

This doesn't look like a permanent solution its kind of temporary. It would be better if Rasa team fix this bug instead of making domain realize by adding utter_chitchat or utter_faq.
I hope you understand

**Comment by Rohitdk88** [comment]
I am using rasa after months. First time into rasa 2+

I was trying to migrate my previous bot explained by Mady mantha in "rasa for beginners" course using this blog https://blog.rasa.com/updating-to-rasa-open-source-2-0/

I am facing same issue when form asks about slots.

**Comment by muximuxi** [comment]
same error

**Comment by muximuxi** [comment]
> same error

before training the rasa core , it seems should train the nlu, and use "rasa train -c config_file -d domain_file --data you_NLU_data" instead "rasa train core ", it will be work

**Comment by PsyQuant** [comment]
I am also having this issue - moving my responses to nlu.yml didn't help.
Rasa Version      :         2.7.0
Minimum Compatible Version: 2.6.0
Rasa SDK Version  :         2.7.0
Rasa X Version    :         None
Python Version    :         3.8.0

**Comment by BSlience** [comment]
same issue 
Rasa Version      :         2.8.0
Minimum Compatible Version: 2.8.0
Rasa SDK Version  :         2.8.2
Rasa X Version    :         None
Python Version    :         3.7.11

**Comment by hsm207** [comment]
same issue

Rasa Version      :         2.8.14
Minimum Compatible Version: 2.8.9
Rasa SDK Version  :         2.8.2
Rasa X Version    :         0.42.6
Python Version    :         3.8.10

**Comment by tayfun** [comment]
Fixes have been merged to 3.0.x, main and 2.0.x branches. Will be released in the next micros.
